lib/generators/declare_schema/migration/migrator.rb

```diff
@@ -451,31 +451,68 @@ def change_indexes(model, old_table_name, to_rename)
               end
           end
 
-          drop_indexes = (existing_indexes_without_primary_key - model_indexes_without_primary_key).map do |i|
+          indexes_to_drop = existing_indexes_without_primary_key - model_indexes_without_primary_key
+          indexes_to_add = model_indexes_without_primary_key - existing_indexes_without_primary_key
+
+          renamed_indexes_to_drop = []
+          renamed_indexes_to_add = []
+
+          indexes_to_drop.each { |index_to_drop|
+            renamed_columns = index_to_drop.columns.map do |column|
+              to_rename[column]
+            end
+
+            indexes_to_add.each { |index_to_add|
+              if renamed_columns == index_to_add.columns
+                renamed_indexes_to_drop.append(index_to_drop)
+                renamed_indexes_to_add.append(index_to_add)
+              end
+            }
+          }
```

Comment:
This is where I check if any add or drop index changes are due to a column rename. In this snippet, `columns` is an array of column names and `to_rename` is a hash that maps renaming changes.